### PR TITLE
Always put a digit to left of the decimal separator.

### PR DESCRIPTION
**Pull request by Hacklin**
Show 0.7 instead of .7.
Natural languages always have at least 1 digit to the left
of the decimal separator; unlike some computer languages like C/C++.
Fix issue #3611

### CODE REVIEW

**Comment by jafdy** [comment]
I think this is a good change but you comment is a bit presumptuous, I don't say "zero point seven", I say "point seven", I also write .7 not 0.7 and typically type it too. So, good readability change but there isn't always a digit left of the point.

**Comment by Hacklin** [comment]
Except for computer languages I've never seen it written as .7 always as 0.7.
And in Dutch and German you'd also say 0,7 and not ,7.

Could you point me to a publication in which they omit the 0 before the dot? (i.e. write it as .7)

**Comment by tehhowch** [comment]
I believe jafdy is speaking in terms of intermediate work, and not formal printed publications, i.e. when conversing with coworkers, family, etc

**Comment by Hacklin** [comment]
Ah.
I know it's done in english spoken text,
so it would make sense to write .7 in mission conversations.
But we're talking about descriptions of ships and outfits.

In other words it depends on where the number is being used:
- 0.7 in descriptions of ships, outfits, items,
-  .7 in spoken conversations,
- 0.7 in the HUD / main window ??
- always 0,7 for languages that use a comma as decimal separator
  (just in case ES gets translated).

I could add an extra parameter to the format function.

**Comment by Tadrix** [comment]
0.7 is a good change in my opinion.

There's no really good way to cover _spoken_ conversations AFAIK. At least I heard (admittedly in SF movies and shows most of the time) both "point-seven" and "zero-point-seven", and exact wording should be left for mission writers to decide on.

However, where the game represents numbers numerically, they probably should be represented as in formal writing (at least where it is handy for them to be standardized).

File: source/Format.cpp [hunk 1]
```diff
@@ -101,6 +105,7 @@ string Format::Number(double value)
 				result += ',';
 		}
 	}
+	while(rounded || right);
```
Comment (tehhowch) [inline source/Format.cpp]: For `do {} while` loops in Endless Sky code, the `while` should be on the same line as the closing `do` brace.

Comment (Hacklin) [reply source/Format.cpp]: Thanks.

**Pull request merged**